### PR TITLE
Parameter Namespacing: Refactoring using the ParameterPath object.

### PR DESCRIPTION
**Pull request by AAlon**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ross-desmond** [comment]
Requires major version bump

**Comment by AAlon** [comment]
Addressed feedback.

File: aws_ros1_common/test/parameter_reader_test.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    ASSERT_EQ(std::string(), complex_no_node_ns_result);
+    consumer.setParam(PARAM_READER_TEST__PARAM_PREFIX "/" PARAM_READER_TEST__PARAM_KEY, PARAM_READER_TEST__PARAM_VALUE);
+    parameter_reader->ReadStdString(param_path_complex_no_node_ns, complex_no_node_ns_result);
+    ASSERT_EQ(complex_no_node_ns_result, complex_with_node_ns_result);
+}
+
+TEST(ParameterReader, failureTests)
+{
+    auto parameter_reader = std::make_shared<Aws::Client::Ros1NodeParameterReader>();
+    auto inexistent_path = ParameterPath("I don't exist");
```
Comment (mm318) [inline aws_ros1_common/test/parameter_reader_test.cpp]: `nonexistent_path` :)

File: aws_ros1_common/test/parameter_reader_test.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    std::string inexistent_path_result = PARAM_READER_TEST__PARAM_VALUE;
+    /* Querying for a nonexistent parameter should return NOT_FOUND and the out parameter remains unchanged. */
+    ASSERT_EQ(Aws::AwsError::AWS_ERR_NOT_FOUND, parameter_reader->ReadStdString(inexistent_path, inexistent_path_result));
```
Comment (mm318) [inline aws_ros1_common/test/parameter_reader_test.cpp]: Nonexistent seems to be an understatement. It is an invalid format for a path. I wonder if we should still be returning `AWS_ERR_NOT_FOUND` in the case of an invalid path.

**Pull request merged**